MediaBrowser.Model/Cryptography/PasswordHash.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MediaBrowser.Model.Cryptography
+{
+    public class PasswordHash
+    {
+        //Defined from this hash storage spec
+        //https://github.com/P-H-C/phc-string-format/blob/master/phc-sf-spec.md
+        //$<id>[$<param>=<value>(,<param>=<value>)*][$<salt>[$<hash>]]
+
+        public string Id;
+        public Dictionary<string, string> Parameters = new Dictionary<string, string>();
+        public string Salt;
+        public byte[] SaltBytes;
+        public string Hash;
+        public byte[] HashBytes;
+        public PasswordHash(string StorageString)
+        {
+            string[] a = StorageString.Split('$');
+            Id = a[1];
+            if (a[2].Contains("="))
+            {
+                foreach (string paramset in (a[2].Split(',')))
+                {
+                    if (!String.IsNullOrEmpty(paramset))
+                    {
+                        string[] fields = paramset.Split('=');
+                        Parameters.Add(fields[0], fields[1]);
+                    }
+                }
+                if (a.Length == 4)
+                {
+                    Salt = a[2];
+                    SaltBytes = FromByteString(Salt);
+                    Hash = a[3];
+                    HashBytes = FromByteString(Hash);
+                }
+                else
+                {
+                    Salt = string.Empty;
+                    Hash = a[3];
+                    HashBytes = FromByteString(Hash);
+                }
+            }
+            else
+            {
+                if (a.Length == 4)
+                {
+                    Salt = a[2];
+                    SaltBytes = FromByteString(Salt);
+                    Hash = a[3];
+                    HashBytes = FromByteString(Hash);
+                }
+                else
+                {
+                    Salt = string.Empty;
+                    Hash = a[2];
+                    HashBytes = FromByteString(Hash);
+                }
+
+            }
+
+        }
+
+        public PasswordHash(ICryptoProvider cryptoProvider2)

```

Comment:
why `2` in parameter name?

Reply:
because I was working on this side by side and forgot to update the reference when I replaced the defaults rather than making second versions. Fixed